packages/strapi-generate-model/lib/before.js

```diff
@@ -34,41 +35,32 @@ module.exports = (scope, cb) => {
     );
   }
 
+  // Format `id`.
+  const name = scope.name || nameToSlug(scope.id);
+
   // `scope.args` are the raw command line arguments.
   _.defaults(scope, {
-    id: _.trim(_.deburr(scope.id)),
+    name,
     idPluralized: pluralize.plural(_.trim(_.deburr(scope.id))),
```

Comment:
Now that `nameToCollectionName`  is in strapi-utils you can use it to create the collectionName / idPluralized. you can refactor to get ride of idPluralized and only use collectionName ;)